x-pack/plugins/infra/common/alerting/logs/log_threshold/types.ts

```diff
@@ -301,4 +301,58 @@ export const GroupedSearchQueryResponseRT = rt.intersection([
   }),
 ]);
 
+export type UnoptimizedGroupedSearchQueryResponse = rt.TypeOf<
+  typeof UnoptimizedGroupedSearchQueryResponseRT
+>;
+
+export const OptimizedGroupedSearchQueryResponseRT = rt.intersection([
```

Comment:
Between the optimised and unoptimised types there are only a few things that change, however I found it more readable to separate them out fully. Maybe just chunks like the preview charts could be reusable types. Opinions welcome of course 😄

Reply:
👍 I'll have a little tweak of the types and see if I can make this more obvious / reusable.